packages/filesystem/src/node/nsfw-watcher/nsfw-filesystem-watcher.ts

```diff
@@ -175,14 +175,13 @@ export class NsfwFileSystemWatcherServer implements FileSystemWatcherServer {
         const uri = FileUri.create(path).toString();
         this.changes.push({ uri, type });
 
-        this.toDisposeOnFileChange.dispose();
-        const timer = setTimeout(() => this.fireDidFilesChanged(), this.fireDidFilesChangedTimeout);
-        this.toDisposeOnFileChange.push(Disposable.create(() => clearTimeout(timer)));
+        this.fireDidFilesChanged();
     }
 
-    protected fireDidFilesChanged(): void {
-        const changes = this.changes;
-        this.changes = [];
+    protected readonly fireDidFilesChanged: () => void = debounce(() => this.doFireDidFilesChanged(), 50);
```

Comment:
Some comments around here would help too, to describe what is the intended behavior.